src/coreclr/jit/emitriscv64.cpp

```diff
@@ -4316,25 +4314,24 @@ regNumber emitter::emitInsTernary(instruction ins, emitAttr attr, GenTree* dst,
 
                     if (needCheckOv)
                     {
-                        assert(codeGen->rsGetRsvdReg() != dst->GetRegNum());
-                        assert(codeGen->rsGetRsvdReg() != src1->GetRegNum());
-                        assert(codeGen->rsGetRsvdReg() != src2->GetRegNum());
+                        assert(tempReg != dst->GetRegNum());
+                        assert(tempReg != src1->GetRegNum());
+                        assert(tempReg != src2->GetRegNum());
 
                         if ((dst->gtFlags & GTF_UNSIGNED) != 0)
                         {
-                            codeGen->genJumpToThrowHlpBlk_la(SCK_OVERFLOW, INS_bne, codeGen->rsGetRsvdReg());
+                            codeGen->genJumpToThrowHlpBlk_la(SCK_OVERFLOW, INS_bne, tempReg);
                         }
                         else
                         {
-                            regNumber tmpReg = dst->GetSingleTempReg();
-                            assert(tmpReg != dst->GetRegNum());
-                            assert(tmpReg != src1->GetRegNum());
-                            assert(tmpReg != src2->GetRegNum());
+                            regNumber tempReg2 = dst->ExtractTempReg();
```

Comment:
It is but tempReg is to replace rsGetRsvdReg and tempReg2 is another temp that was assigned here before (I just renamed it from tmpReg to avoid confusion).

Reply:
Sure, but isn't tempReg2 always equal to tempReg? If so we can simply use tempReg and remove tempReg2 completely. I don't think dst is mutated somewhere in-between but maybe I'm missing something.

Reply:
Both are used in genJumpToThrowHlpBlk_la below so probably not.